plugins/outputs/nats/nats_test.go

```diff
@@ -263,3 +266,230 @@ func createStream(t *testing.T, server string, cfg *nats.StreamConfig) {
 	_, err = js.AddStream(cfg)
 	require.NoError(t, err)
 }
+func TestWriteWithLayoutIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	natsServicePort := "4222"
+	type testConfig struct {
+		name                    string
+		container               testutil.Container
+		nats                    *NATS
```

Comment:
Please do not define the whole plugin here but only the parameter you want to vary in the test.